fix(core): log error instead of warning for unknown properties and elements

crisbeto opened the pull request:
**Note:** this is a resubmit of #35798 intended for v10.

Changes the Ivy unknown element/property messages from being logged with `console.warn` to `console.error`. This should make them a bit more visible without breaking existing apps. Furthermore, a lot of folks filter out warning messages in the dev tools' console, whereas errors are usually still shown.

BREAKING CHANGE:
Warnings about unknown elements are now logged as errors. This won't break your app, but it may trip up tools that expect nothing to be logged via `console.error`.

Fixes #35699.

atscott commented:
Per comment on previous PR, https://github.com/angular/angular/pull/35798#issuecomment-594129821, this will need a g3 patch when merged (cl/298655911).

crisbeto committed:
fix(core): log error instead of warning for unknown properties and elements

Changes the Ivy unknown element/property messages from being logged with `console.warn` to `console.error`. This should make them a bit more visible without breaking existing apps. Furthermore, a lot of folks filter out warning messages in the dev tools' console, whereas errors are usually still shown.

BREAKING CHANGE:
Warnings about unknown elements are now logged as errors. This won't break your app, but it may trip up tools that expect nothing to be logged via `console.error`.

Fixes #35699.

IgorMinar reviewed (commented):
This looks reasonable to me for v10, but I'll defer to other FW folks.

pkozlowski-opensource reviewed (approved):
LGTM

kara reviewed (approved):
LGTM

Reviewed-for: global-approvers

kara commented:
[presubmit](http://test/OCL:309488757:BASE:309488777:1588373440701:9ea80c57)

kara commented:
Caretaker note: Merging this into G3 requires a patch because there is a test that checks exactly which console method we call (see [here](https://cs.corp.google.com/piper///depot/google3/javascript/angular2/testing/catalyst/integration_test.ts?l=169&ws=tap-prod-presubmit/74523051&snapshot=2))
